### DISCUSSION TITLE
The necessity to expose subcomponent factory method in parent component

### DISCUSSION THREAD

**Discussion by kevinguitar** [ZacSweers/anvil#66] [Q&A] [answered]
Hi Zac,

Currently, we have to expose the subcomponent factory in the parent component (example attached below), but looking at the dagger-generated component (`DaggerMerged*Component`), the method is actually not used.
https://github.com/kevinguitar/anvil-issue-replicate/blob/main/app/src/main/java/com/example/anvilissuereplicate/MainActivitySubcomponent.kt#L37-L40

I think it looks related to this section, but I'm not entirely sure what will be the next step, will Anvil take care of this factory exposure, or it's necessary to declare it in code? Thank you 🙏 
https://github.com/ZacSweers/anvil/blob/main/FORK.md#a-note-about-subcomponent-factory-methods

Replicate the issue here:
https://github.com/kevinguitar/anvil-issue-replicate/pull/2

**Comment by ZacSweers** [comment]
This seems to be working as intended? In your replication branch, you removed the contributed interface that dagger needs to see it no? The code you linked first is not dagger-generated code so I'm not sure I follow.

**Reply by kevinguitar** [reply]
> you removed the contributed interface that dagger needs to see it no?

Ah ok, this is what I wanted to know, so dagger requires explicit Factory contribution to parent component now. I was confused because it wasn't required before (with anvil upstream), we can specify the `subcomponents` in the module that we provide the factory through multi-binding, and it will just work.
```kotlin
@Module(subcomponents = [MainActivitySubcomponent::class])
@ContributesTo(scope = AppScope::class)
interface MainActivityInjectorBinder {
    @IntoMap
    @Binds
    @ClassKey(MainActivity::class)
    fun bindMainActivityAnvilInjector(`impl`: MainActivitySubcomponent.Factory):
            AnvilAndroidInjector.Factory<*>
}

```

**Reply by ZacSweers** [reply, marked as answer]
it doesn't have to be a factory, it can be the subcomponent itself too. It was always required in some form, how else would you know what to access to retrieve an instance?

**Reply by kevinguitar** [reply]
Thanks! It's pretty clear now 🙏
